extensions/wikia/AdEngine/js/provider/revcontent.js

```diff
@@ -0,0 +1,70 @@
+/*global define*/
+define('ext.wikia.adEngine.provider.revcontent', [
+	'wikia.document',
+	'wikia.log',
+	'wikia.window'
+], function (doc, log, win) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.provider.revcontent',
+		slotMap = {
+			REVCONTENT_ABOVE_ARTICLE: { rslotId: 'rcjsload_wikiaaa', widgetId: 30217 },
+			REVCONTENT_RIGHT_RAIL:	  { rslotId: 'rcjsload_wikiarr', widgetId: 30231 },
+			REVCONTENT_BELOW_ARTICLE: { rslotId: 'rcjsload_wikiaba', widgetId: 30229 }
+		};
+
+	function canHandleSlot(slotName) {
+		var res = !!slotMap[slotName];
+		log(['canHandleSlot', slotName, res], 'debug', logGroup);
+		return res;
+	}
+
+	function fillInSlot(slot) {
+		log(['fillInSlot', slot.name], 'debug', logGroup);
+
+		var slotInfo = slotMap[slot.name];
+		addWidget(slotInfo.widgetId, slotInfo.rslotId, slot.name);
+
+		log(['fillInSlot', slot.name, 'done'], 'debug', logGroup);
+	}
+
+	function addWidget(widgetId, rslotId, slotName) {
+		var rparent,
+			rslot,
+			rscript;
+
+		rslot = doc.createElement('div');
+		rslot.id = rslotId;
+
+		rscript = doc.createElement('script');
+		rscript.type = 'text/javascript';
+		rscript.id = 'rc_' + Math.floor(Math.random() * 1000);
+		rscript.src = genSrc(rscript.id, widgetId);
```

Comment:
I use JSHint and according to the config you can use function before it was defined: https://github.com/Wikia/app/blob/dev/.jshintrc#L13